gnomad_qc/v3/sample_qc/sample_qc.py

```diff
@@ -534,6 +631,17 @@ def assign_pops(
     if pop is not None:
         meta_ht = meta.ht()
         pop_pca_scores_ht = pop_pca_scores_ht.annotate(**meta_ht[pop_pca_scores_ht.key])
+        # Set subpop_descritpion to missing if the subpop is not in the manually curated list of subpops for the given pop
+        pop_pca_scores_ht = pop_pca_scores_ht.annotate(
+            project_meta=pop_pca_scores_ht.project_meta.annotate(
```

Comment:
Is `subpop_description` all you need from the `meta_ht`?

If so, I would actually just use the project meta, so I think this line below can be moved outside the if statement

```
project_meta_ht = project_meta.ht()[pop_pca_scores_ht.key]
```

Then instead of

```
pop_pca_scores_ht = pop_pca_scores_ht.annotate(**meta_ht[pop_pca_scores_ht.key])
```
do
```
pop_pca_scores_ht = pop_pca_scores_ht.annotate(
                    subpop_description=hl.or_missing(
                    hl.literal(CURATED_SUBPOPS[pop]).contains(
                        pop_pca_scores_ht.subpop_description
                    ),
                    pop_pca_scores_ht.subpop_description,
                )
    )
```

and change uses of `pop_pca_scores_ht.project_meta.subpop_description` to `pop_pca_scores_ht.subpop_description`

Reply:
I think you might want to also add the curated subpops as an optional list parameter to the function then within the subpops script pass in `curated_subpops = CURATED_SUBPOPS[pop]`. Then if passed in use it otherwise just use `pop_pca_scores_ht.subpop_description`